src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -115,16 +113,180 @@ public CodeGenIL(IDictionary<string, IList> input, string filePath)
 
         private Type EmitCoercionCode(Type argType, ParameterInfo param)
         {
+            if (param.ParameterType.IsAssignableFrom(argType)) return argType;
+
             if(argType == typeof(double) && param.ParameterType == typeof(long))
             {
                 EmitOpCode(OpCodes.Conv_I8);
                 return typeof(long);
             }
+            if (argType == typeof(double) || argType == typeof(long) && param.ParameterType == typeof(int))
```

Comment:
maybe enclose the argType conditions in parenthesis ?
if ( `(`argType == typeof(double) || argType == typeof(long) `)` &&